client/packages/programs/src/JsonForms/common/components/Array.tsx

```diff
@@ -254,7 +297,113 @@ const ArrayComponent = (props: ArrayControlCustomProps) => {
     label,
     rootSchema,
     renderers,
+    config,
   } = props;
+  const { errors: zErrors, options } = useZodOptionsValidation(
+    Options,
+    props.uischema.options
+  );
+
+  const defaultExpanded = props.uischema?.options?.['defaultExpanded'] ?? false;
+
+  const [expandedItems, setExpandedItems] = useState<boolean[]>(
+    new Array(inputData?.length ?? 0).fill(defaultExpanded)
+  );
+
+  const isNotesArray = options?.detail?.elements[0]?.type === 'Note';
+
+  const isElementEditable = (child: any, index: number) => {
+    if (!enabled) return false;
+    if (!options?.editRestrictions) return true;
+
+    const restrictions = options.editRestrictions;
+
+    // Must be editable if no timestamp (means it's just created)
+    if (!child.created) return true;
+
+    // Only allow the latest element to be edited
+    if (restrictions?.latest && index !== data.length - 1) return false;
+
+    // Author must match the current user
+    if (
+      restrictions?.isCurrentUser &&
+      child?.authorId &&
+      child.authorId !== config.user.id
+    )
+      return false;
+
+    // Must not be older than `maxAge` days
+    if (DateUtils.ageInDays(child.created) >= 1) return false;
+
+    return true;
+  };
+
+  const getItemLabel = (child: any, index: number) => {
+    const isExpanded = expandedItems[index];
+
+    // For most arrays, or Notes with and explicit itemLabel pattern
+    if (uischema.itemLabel || !isNotesArray)
+      return (
```

Comment:
Think its better to wrap the `return` statement into brackets {} when it spans more than one line. Find it a bit hard to read...